generatebundlefile/main.go

```diff
@@ -159,17 +232,16 @@ func main() {
 			if err != nil {
 				BundleLog.Error(err, "Unable create find Public ECR URI for destination account")
 			}
-			dockerReleaseStruct := &DockerAuth{
+			dockerReleaseStruct = &DockerAuth{
 				Auths: map[string]DockerAuthRegistry{
 					fmt.Sprintf("public.ecr.aws/%s", clients.ecrPublicClient.SourceRegistry): DockerAuthRegistry{clients.ecrPublicClient.AuthConfig},
 					"public.ecr.aws": DockerAuthRegistry{clients.ecrPublicClientRelease.AuthConfig},
 				},
 			}
-			authFile, err := NewAuthFile(dockerReleaseStruct)
+			authFile, err = NewAuthFile(dockerReleaseStruct)
 			if err != nil || authFile == "" {
 				BundleLog.Error(err, "Unable create AuthFile")
 			}
-			defer os.Remove(authFile)
```

Comment:
Seems like we should be particularly carfule removing these auth files here and those NewAuthFiel calls above or am I missing something. Can't we wrap this in an object and change NewAuthFile to NewAuthenticator that managed this object and you call `defer authenticator.Close()` which would do this os.Remove call?

Reply:
Yea that seems likes a good idea, I'll make that change.

Reply:
I added the Close method mentioned in the comment in [09d9ca5](https://github.com/aws/eks-anywhere-packages/pull/247/commits/09d9ca5e659b456a0ba9ae41c655fe2f78ef1cac)